### DISCUSSION TITLE
Crontab rewrite with sections causes migration errors

### DISCUSSION THREAD

**Discussion by SimJoSt** [deployphp/deployer#3895] [Bugs]
Migrated from issue https://github.com/deployphp/deployer/issues/3768

### Deployer Version

v7.3.3

### Target OS

Ubuntu 22.04

### Which PHP version are you using?

PHP 8.2

### Content of deploy.php or deploy.yaml

```yaml
import:
  - contrib/crontab.php

config:
  crontab:identifier: 'application4'
  crontab:jobs:
      - '* * * * * cd {{current_path}} && {{bin/php}} wp-cron.php >> /dev/null 2>&1'

hosts:
  ***redacted***:
    deploy_path: '~/application4'
```

### Steps to reproduce

We've been using the `crontab` recipe in our config for some time now, and I welcome the [rewrite with sections](https://github.com/deployphp/deployer/pull/3708), as it prevents duplicate entries when the crontab config is changed and resynced.
Without initially knowing about the change, it broke some of our applications and no crontab was running for some time. Only the terminal output messages, pointed me towards the new behavior. I get that this is only a change in a smaller recipe, but it seems to be a breaking change.

If there is a migration in place, it shouldn't be an issue. However, I noted some issues while using it and migrating existing applications to it.
Running the `crontab:sync` task showed it recognized a previous job in the crontab config on the server, said it would move it to the section, said it couldn't find the section, and said it would create the section.
The section was created, and the previous job was removed. However, the pre-existing job was not created anew in the new section. It seems like it cannot perform the migration and creation of a new section at the same time, breaking existing configurations.
The fix is to run the `crontab:sync` task again, which fixes the problem.

Before:
```bash
➜  dep ssh
➜  crontab -l
* * * * * cd ~/application1/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
* * * * * cd ~/application2/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
* * * * * cd ~/application3/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
* * * * * cd ~/application4/current && /usr/bin/php8.2 wp-cron.php >> /dev/null 2>&1
```

Migration:
```bash
➜  dep crontab:sync
task crontab:sync
[***redacted***] Crontab: Found existing job in crontab, moving it to the section
[***redacted***] Crontab: Found no section, created the section with configured jobs
➜  dep ssh
➜  crontab -l
* * * * * cd ~/application1/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
* * * * * cd ~/application2/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
* * * * * cd ~/application3/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
###< application4
###> application4
```

Re-run:
```bash
➜  dep crontab:sync
➜  dep ssh
➜  crontab -l
* * * * * cd ~/application1/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
* * * * * cd ~/application2/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
* * * * * cd ~/application3/current && /usr/bin/php8.1 artisan schedule:run >> /dev/null 2>&1
###< application4
* * * * * cd ~/application4/current && /usr/bin/php8.2 wp-cron.php >> /dev/null 2>&1
###> application4
```

**Comment by SimJoSt** [comment]
The issue already contained a few comments, especially by @ardentsword, that are relevant to the status.

**Comment by antonmedv** [comment]
so the problem only with migration from previous recipe to current?

**Reply by antonmedv** [reply]
And the current recipe is working as expected?

**Reply by SimJoSt** [reply]
Yes, exactly. If you start fresh or rerun the `crontab` task in general, there are no issues.

**Reply by antonmedv** [reply]
We need better e2e tests.
